src/modules/compliance/src/lib/procedures/EnsureInteractiveUsersHomeDirectoriesAreConfigured.cpp

```diff
@@ -0,0 +1,193 @@
+            {
+                OsConfigLogError(context.GetLogHandle(), "Failed to stat home directory '%s' for user '%s': %s", pwd->pw_dir, pwd->pw_name, strerror(status));
+                return Error(string("Failed to stat home directory: ") + strerror(status), status);
+            }
+        }
+
+        const auto* group = getgrgid(pwd->pw_gid);
+        if (nullptr == group)
+        {
+            OsConfigLogError(context.GetLogHandle(), "Failed to get group for user '%s': %s", pwd->pw_name, strerror(errno));
+            return Error(string("Failed to get group for user: ") + strerror(errno), errno);
+        }
+
+        map<string, string> arguments = {{"filename", pwd->pw_dir}, {"mask", "027"}, {"owner", pwd->pw_name}, {"group", group->gr_name}};
+        indicators.Push("EnsureFilePermissions");
+        auto subResult = AuditEnsureFilePermissions(std::move(arguments), indicators, context);
+        indicators.Pop();
+        if (!subResult.HasValue())
+        {
+            OsConfigLogError(context.GetLogHandle(), "Failed to check permissions for home directory '%s' for user '%s': %s", pwd->pw_dir, pwd->pw_name,
+                subResult.Error().message.c_str());
+            return subResult.Error();
+        }
+
+        if (subResult.Value() == Status::NonCompliant)
+        {
+            OsConfigLogInfo(context.GetLogHandle(), "User '%s' has home directory '%s' with incorrect permissions", pwd->pw_name, pwd->pw_dir);
+            return indicators.NonCompliant(std::string("User's '") + pwd->pw_name + "' home directory '" + pwd->pw_dir + "' has incorrect permissions");
+        }
+
+        return Status::Compliant;
+    };
+
+    return IterateUsers(cb, BreakOnNonCompliant::False, context);
+}
+
+REMEDIATE_FN(EnsureInteractiveUsersHomeDirectoriesAreConfigured)
+{
+    UNUSED(args);
+
+    const auto validShells = ListValidShells(context);
+    if (!validShells.HasValue())
+    {
+        OsConfigLogError(context.GetLogHandle(), "Failed to get valid shells: %s", validShells.Error().message.c_str());
+        return validShells.Error();
+    }
+
+    auto cb = [&validShells, &indicators, &context](const passwd* pwd) -> Result<Status> {
+        const auto shell = string(pwd->pw_shell);
+        const auto it = validShells->find(shell);
+        if (it == validShells->end())
+        {
+            OsConfigLogDebug(context.GetLogHandle(), "User '%s' has shell '%s' not in /etc/shells", pwd->pw_name, pwd->pw_shell);
+            return Status::Compliant;
+        }
+
+        struct stat st;
+        if (stat(pwd->pw_dir, &st) != 0)
+        {
+            int status = errno;
+            OsConfigLogDebug(context.GetLogHandle(), "stat failed for home directory '%s' for user '%s': %s", pwd->pw_dir, pwd->pw_name, strerror(status));
+            if (status == ENOENT)
+            {
+                // Home directory does not exist, so we need to create it
+                if (0 != mkdir(pwd->pw_dir, 0750))
+                {
```

Comment:
CIS says the remediation may depend on site-policy:
```
• Lock the user account 
• Remove the user from the system 
• create a directory for the user. If undefined, edit /etc/passwd and add the 
absolute path to the directory to the last field of the user.
```

This warrants at least documenting the behavior somewhere and capturing a TODO.